sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -137,6 +137,8 @@ object TypeCoercion {
     case (DateType, TimestampType) => Some(StringType)
     case (StringType, NullType) => Some(StringType)
     case (NullType, StringType) => Some(StringType)
+    case (n: DecimalType, s: StringType) => Some(DoubleType)
```

Comment:
add a comment to say that there is no proper decimal type we can pick, using double type is the best we can do.